book/integration-service.md

```diff
@@ -0,0 +1,271 @@
+
+**Ephemeral Environment** - The ephemeral copy of an existing environment that’s created by the Integration service for use during a specific Integration pipeline run. After the environment is copied and marked as ephemeral, the Integration service creates a new SnapshotEnvironmentBinding linking the Snapshot under test with the newly created Environment. This triggers a deployment of the Snapshot on the new Environment, handled by the GitOps service. Finally, the Integration service links the credentials for the Environment to the Integration Tekton pipeline run and starts the test run.
+
+
+## Resources 
+Below are the list of CR’s that the integration service is responsible for interacting and performing operations:
+
+### CREATE
+
+| Custom Resource | When? | Why? |
+|---|---|---|
+| Snapshot | Post Component build PipelineRun completes.  | Used as the source input for the test gate pipeline |
+| PipelineRuns | Post Component build PipelineRun completes  | To test the Snapshot  |
+| Release | Post PipelineRun if there is a ReleaseLink for the Application, if the auto-release flag is set on the ReleaseLink | To signal to the Release Service to automatically push the Snapshot to the Production Environment |
+
+### READ
+
+| Custom Resources | When? | Why? |
+|---|---|---|
+| Application & Component | Before creating the Snapshots | To know which build.ImageSpecs to use in the Snapshot |
+| IntegrationTestScenario | Before creating the Component/Composite PipelineRun(s) | To get the information for the PipelineRun - which Tekton bundles and Environments will be used |
+| ReleaseLink | Before creating the Release | To signal the Release Service for next environment promotion (prod) |
+
+### UPDATE
+
+| Custom Resource | When? | Why? |
+|---|---|---|
+| SnapshotEnvironmentBinding | Post Test PipelineRun | To promote the Snapshot to the next environment (Stage/Stable) |
+| Snapshot | Post Test PipelineRun | To mark the snapshot validated so that it can be promoted to the next environment |
+| Component | Post Test PipelineRun | To update the Component with the image spec, in order to update the global candidates list |
+
+### WATCH
+
+| Custom Resource | When? | Why? |
+|---|---|---|
+| PipelineRun | Post Build PipelineRun | Read the component and the output image to create the Snapshot and start the Preliminary PipelineRun |
+| PipelineRun | Post Preliminary PipelineRun | Check the result of the Preliminary PipelineRun and  follow up on the global candidate list state to either trigger the Final PipelineRun or check ReleaseLink and create a Release |
+| PipelineRun | Post Final PipelineRun | Check the result of the Final PipelineRun, check ReleaseLink and create a Release |
+
+### Pipeline
+
+The test pipeline will consist of a single definition that could be executed as two logically distinct pipeline runs by the Integration Service. The logical pipeline runs are:
+
+- Component (Prelim) pipeline run
```

Comment:
Remove the deprecated terminology here. It is old enough that we don't need to keep mentioning it down here.

Reply:
Done